plugins/dell-dock/fu-dell-dock-hid.c

```diff
@@ -71,12 +71,14 @@ fu_dell_dock_hid_set_report (FuDevice *self,
 		    G_USB_DEVICE_RECIPIENT_INTERFACE, HID_REPORT_SET, 0x0200,
 		    0x0000, outbuffer, 192, &actual_len,
 		    HIDI2C_TRANSACTION_TIMEOUT, NULL, &error_local);
-		if (ret)
+		if (ret) {
 			break;
-		if (i == HID_MAX_RETRIES ||
-		    g_error_matches (error_local,
-				     G_USB_DEVICE_ERROR,
-				     G_USB_DEVICE_ERROR_NO_DEVICE)) {
+		} else if (i == HID_MAX_RETRIES) {
```

Comment:
Aha, yes, this is the bug I saw. i == HID_MAX_RETRIES, and there was nothing in `error_local` to propagate when returning FALSE.